utility/servicer/module.go

```diff
@@ -182,31 +356,107 @@ func (s servicer) admitRelay(relay *coreTypes.Relay) error {
 
 	height := s.GetBus().GetConsensusModule().CurrentHeight()
 	if err := s.validateRelayMeta(relay.Meta, int64(height)); err != nil {
-		return fmt.Errorf("%w: %s", errValidateRelayMeta, err.Error())
+		return fmt.Errorf("%s: %w", err.Error(), errValidateRelayMeta)
 	}
 
 	// TODO: update the CLI to include ApplicationAddress(or Application Public Key) in the RelayMeta
-	session, err := s.GetBus().GetUtilityModule().GetSession(relay.Meta.ApplicationAddress, int64(height), relay.Meta.RelayChain.Id, relay.Meta.GeoZone.Id)
+	session, err := s.getSession(relay)
 	if err != nil {
-		return fmt.Errorf("%s: failed to get a session for height %d for relay meta %s: %w", errPrefix, height, relay.Meta, err)
+		return err
 	}
 
 	// TODO: (REFACTOR) use a loop to run all validators: would also remove the need for passing the session around
 	if err := validateRelayBlockHeight(relay.Meta, session); err != nil {
-		return fmt.Errorf("%w: %s", errValidateBlockHeight, err.Error())
+		return fmt.Errorf("%s: %w", err.Error(), errValidateBlockHeight)
 	}
 
-	if err := s.validateApplication(relay.Meta, session); err != nil {
-		return fmt.Errorf("%s: relay failed application validation: %w", errPrefix, err)
+	if err := s.validateServicer(relay.Meta, session); err != nil {
+		return fmt.Errorf("%s: %s: %w", errPrefix, err.Error(), errValidateServicer)
 	}
 
-	if err := s.validateServicer(relay.Meta, session); err != nil {
-		return fmt.Errorf("%s: relay failed servicer instance validation: %w", errPrefix, err)
+	if err := s.validateApplication(session); err != nil {
+		return fmt.Errorf("%s: %s: %w", errPrefix, err.Error(), errValidateApplication)
 	}
 
 	return nil
 }
 
+// DISCUSS: do we need to export this functionality as part of the utility module?
```

Comment:
I think we can remove this `DISCUSS` item unless you see a reason for it. 

My answer is no.